acceptancetests/deploy_stack.py

```diff
@@ -570,6 +571,40 @@ def tear_down(self, has_controller):
             self.tear_down_client.kill_controller(check=True)
 
 
+class ExistingController:
+    """A Controller strategy where the controller is already present.
+
+    Intended for use with BootstrapManager.
+    """
+
+    def __init__(self, client, tear_down_client, juju_data_dir, env):
+        self.client = client
+        self.client.env.environment = env
+        self.tear_down_client = tear_down_client
+
+    def create_initial_model(self, upload_tools):
+        """Create the initial model."""
+        self.client.add_model(self.client.env)
+        logging.info('Added model {} to existing controller'.format(
+            self.client.env.environment))
+
+    def prepare(self):
+        """Prepare client for use by killing the existing controller."""
+        self.tear_down_client.kill_controller()
```

Comment:
Killing the controller seems counter productive when the goal is using an existing, or have a misinterpreted what this actually does?